search/api_test.py

```diff
@@ -1,365 +1,122 @@
-        assert str(ex.exception) == "Unable to find index {}".format(settings.ELASTICSEARCH_INDEX)
-
-    def test_no_mapping(self):
-        """
-        Test that error is raised if we don't have a mapping
-        """
-        conn = get_conn(verify=False)
-        conn.indices.create(settings.ELASTICSEARCH_INDEX)
-
-        with self.assertRaises(ReindexException) as ex:
-            get_conn()
-        assert str(ex.exception) == "Mapping {} not found".format(USER_DOC_TYPE)
-
-
-class RecreateIndexTests(ESTestCase):
-    """
-    Tests for management commands
-    """
-    def setUp(self):
-        """
-        Start without any index
-        """
-        super(RecreateIndexTests, self).setUp()
-        conn = get_conn(verify=False)
-        index_name = settings.ELASTICSEARCH_INDEX
-        if conn.indices.exists(index_name):
-            conn.indices.delete(index_name)
+        with self.assertRaises(NoProgramAccessException):
+            create_search_obj(profile.user)
 
-    def test_create_index(self):  # pylint: disable=no-self-use
+    def test_prepare_and_execute_search(self):
         """
-        Test that recreate_index will create an index and let search successfully
+        Test that a Search object is properly prepared and executed
         """
-        recreate_index()
-        assert es.search()['total'] == 0
+        mock_search_func = Mock(name='execute', return_value=['result1', 'result2'])
+        with patch('search.api.create_search_obj', autospec=True, return_value=None) as mock_create_search_obj:
+            results = prepare_and_execute_search(self.user, search_func=mock_search_func)
+            assert mock_create_search_obj.called
+            assert results == ['result1', 'result2']
 
-    def test_update_index(self):  # pylint: disable=no-self-use
+    def test_all_query_matching_emails(self):  # pylint: disable=no-self-use
         """
-        Test that recreate_index will clear old data and index all profiles
+        Test that a set of search results will yield an expected set of emails
         """
-        recreate_index()
-        program_enrollment = ProgramEnrollmentFactory.create()
-        assert_search(es.search(), [program_enrollment])
-        remove_program_enrolled_user(program_enrollment)
-        assert_search(es.search(), [])
-        # recreate_index should index the program-enrolled user
-        recreate_index()
-        assert_search(es.search(), [program_enrollment])
+        fake_results = FakeSearchResults()
+        mock_execute_search = Mock(spec=execute_search, return_value=fake_results)
+        test_es_page_size = 2
+        with patch('search.api.execute_search', mock_execute_search):
+            results = get_all_query_matching_emails(Search(), test_es_page_size)
+            assert mock_execute_search.call_count == math.ceil(fake_results.hits.total/test_es_page_size)
+            assert results == set([result.email for result in fake_results.results])
```

Comment:
Can you add some assertion to make sure the `search_obj = search_obj.fields('email')` from `ESFullResultEmailSearch` was executed correctly?